lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -278,15 +282,15 @@ sub _stdout_cb {
     $buf .= $chunk;
     $buf =~ s/^(.*?\x0a\x0d?\x0a\x0d?)//s or return;    # false until all headers has been read from the CGI script
     $headers = $1;
-
-    if ($headers =~ /^HTTP/) {
+    if ($headers =~ /^HTTP(?: (\d\d\d))?/) {
+      $c->res->code($1) if $1;
       $c->res->parse($headers);
     }
     else {
-      $c->res->code($headers =~ /Location:/ ? 302 : 200);
+      $c->res->code($1) if $headers =~ /^Status: (\d\d\d)/m;
```

Comment:
If the CGI script is not a NPH script but a CGI script then a [Status](https://tools.ietf.org/html/rfc3875#section-6.3.3) header is described in section 6.3.3 of RFC 3875. I'm not implementing a lot of the extra rules mentioned in the section such as "The script SHOULD check the value of SERVER_PROTOCOL before using HTTP/1.1 status codes." I'm just assuming that the CGI script knows what it's doing.